src/lib/viewers/media/DashViewer.js

```diff
@@ -191,25 +192,24 @@ class DashViewer extends VideoBaseViewer {
     }
 
     /**
-     * Handler for hd video
+     * Given a videoId (e.g. hd video id), enables the track with that video ID
+     * while maintaining the SAME AUDIO as the active track.
      *
      * @private
+     * @param {number} videoId - the id of the video used in the variant (provided by Shaka)
      * @return {void}
      */
-    enableHD() {
-        this.showLoadingIcon(this.hdRepresentation.id);
-        this.player.selectVariantTrack(this.hdRepresentation, true);
-    }
-
-    /**
-     * Handler for sd video
-     *
-     * @private
-     * @return {void}
-     */
-    enableSD() {
-        this.showLoadingIcon(this.sdRepresentation.id);
-        this.player.selectVariantTrack(this.sdRepresentation, true);
+    enableVideoId(videoId) {
+        const tracks = this.player.getVariantTracks();
+        const activeTrack = this.getActiveTrack();
+        const newTrack = tracks.find((track) => track.videoId === videoId && track.audioId === activeTrack.audioId);
+        if (newTrack && newTrack.id !== activeTrack.id) {
+            this.showLoadingIcon(newTrack.id);
+            this.player.selectVariantTrack(newTrack, true);
+        } else {
+            // This should never happen...
+            console.log(`Could not find a track with videoId=${videoId}!`);
```

Comment:
Not sure if I'm allowed to do this? I feel like we should log something though...